### PR TITLE
OP-484 movement and lot fix

### PR DESCRIPTION
**Pull request by mwithi**
See OP-484.

Checks fails until https://github.com/informatici/openhospital-core/pull/393 is not merged

### CODE REVIEW

File: src/main/java/org/isf/medicalstock/gui/MovStockMultipleCharging.java [hunk 1]
```diff
@@ -998,7 +1000,8 @@ public Component getTableCellRendererComponent(JTable table, Object value, boole
 		public Component getTableCellRendererComponent(JTable table, Object value, boolean isSelected, boolean hasFocus, int row, int column) {
 			// First format the cell value as required
 			Component cell = super.getTableCellRendererComponent(table, value, isSelected, hasFocus, row, column);
-			value = formatter.format((Number) value);
+			if (value != null)
+				value = formatter.format((Number) value);
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/medicalstock/gui/MovStockMultipleCharging.java:1004]: The cast to `Number` is not really needed and is redundant.  (SonarLint)

File: src/main/java/org/isf/medicalstockward/gui/WardPharmacyRectify.java [hunk 2]
```diff
@@ -581,8 +581,6 @@ private MedicalWard chooseLot(Medical medical, boolean newLot) {
 					jLabelLotQty.setText(medWard.getQty().toString());
 					jLabelInLot.setVisible(true);
 					selectedLot = medWard.getLot();
-					selectedLot.setQuantity(medWard.getQty().intValue());
-					
 				}
 			} while (drugChooseList == null);
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/medicalstockward/gui/WardPharmacyRectify.java:585]: Looking at the code the variable `drugChooseList` can never be null; see for example lines 536, 540, 562, and 576.  Thus the test of the `while` condition is always `false`.

**Comment by dbmalkovsky** [comment]
Using the CORE code from the other PR this compiles cleanly.

File: src/main/java/org/isf/medicalstock/gui/MovStockMultipleCharging.java (hunk 1 above)
Comment (mwithi) [reply src/main/java/org/isf/medicalstock/gui/MovStockMultipleCharging.java]: Removed

File: src/main/java/org/isf/medicalstockward/gui/WardPharmacyRectify.java (hunk 2 above)
Comment (mwithi) [reply src/main/java/org/isf/medicalstockward/gui/WardPharmacyRectify.java]: Removed

File: src/main/java/org/isf/medicalstock/gui/MovStockMultipleCharging.java [hunk 3]
```diff
@@ -872,7 +872,9 @@ public Object getValueAt(int r, int c) {
 			} else if (c == 8) {
 				return cost;
 			} else if (c == 9) {
-				return cost.multiply(new BigDecimal(total));
+				if (cost != null)
+					return cost.multiply(new BigDecimal(total));
+				else return null;
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/medicalstock/gui/MovStockMultipleCharging.java:877]: strange formatting; no braces and no need for `else` after the `return`

File: src/main/java/org/isf/medicalstock/gui/MovStockMultipleCharging.java [hunk 4]
```diff
@@ -998,7 +1000,8 @@ public Component getTableCellRendererComponent(JTable table, Object value, boole
 		public Component getTableCellRendererComponent(JTable table, Object value, boolean isSelected, boolean hasFocus, int row, int column) {
 			// First format the cell value as required
 			Component cell = super.getTableCellRendererComponent(table, value, isSelected, hasFocus, row, column);
-			value = formatter.format((Number) value);
+			if (value != null)
+				value = formatter.format(value);
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/medicalstock/gui/MovStockMultipleCharging.java:1004]: missing braces here too

File: src/main/java/org/isf/medicalstock/gui/MovStockMultipleDischarging.java [hunk 5]
```diff
@@ -623,13 +623,12 @@ private void stripeLots(ArrayList<Lot> lots) {
...
 					if (aLot.getCode().equals(mov.getLot().getCode())) {
-						int aLotQty = aLot.getQuantity();
+						int aLotQty = aLot.getMainStoreQuantity();
 						int newQty = aLotQty - mov.getQuantity();
-						if (newQty == 0) lotIterator.remove();
-						else {
-							aLot.setQuantity(newQty);
+						if (newQty == 0) 
+							lotIterator.remove();
+						else
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/medicalstock/gui/MovStockMultipleDischarging.java:630]: and here too

File: src/main/java/org/isf/medicalstockward/gui/WardPharmacyRectify.java [hunk 6]
```diff
@@ -565,26 +571,22 @@ private MedicalWard chooseLot(Medical medical, boolean newLot) {
...
-			} while (drugChooseList == null);
+			int ok = JOptionPane.showConfirmDialog(WardPharmacyRectify.this, 
+					panel, 
+					MessageBundle.getMessage("angal.medicalstockward.rectify.lotinformations"), //$NON-NLS-1$ 
+					JOptionPane.OK_CANCEL_OPTION);
+
+			if (ok == JOptionPane.OK_OPTION) {
+				int row = lotTable.getSelectedRow();
+				if (row != -1) medWard = drugChooseList.get(row);
+					else return null;
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/medicalstockward/gui/WardPharmacyRectify.java:582]: strange formatting

**Comment by dbmalkovsky** [comment]
Other than cosmetic issues LGTM

File: src/main/java/org/isf/medicalstock/gui/MovStockMultipleCharging.java (hunk 3 above)
Comment (mwithi) [reply src/main/java/org/isf/medicalstock/gui/MovStockMultipleCharging.java:877]: Thanks!

File: src/main/java/org/isf/medicalstockward/gui/WardPharmacyRectify.java (hunk 6 above)
Comment (mwithi) [reply src/main/java/org/isf/medicalstockward/gui/WardPharmacyRectify.java:582]: Thanks!

**Pull request merged**